models/Document/Page.php

```diff
@@ -274,4 +274,18 @@ public function getPreviewImageFilesystemPath()
     {
         return PIMCORE_SYSTEM_TEMP_DIRECTORY . '/document-page-previews/document-page-screenshot-' . $this->getId() . '@2x.jpg';
     }
+
+    public function save()
+    {
+        // Dispatch page preview message, if preview is enabled.
+        $documentsConfig = \Pimcore\Config::getSystemConfiguration('documents');
+        if ($documentsConfig['generate_preview'] ?? false) {
+
+            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
+                new GeneratePagePreviewMessage($this->getId(), \Pimcore\Tool::getHostUrl())
+            );
+        }
+
+        return parent::save();
```

Comment:
Shouldn't it be sent to the messenger bus after saving? Not that the old page is still generated?